src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,216 @@
+	Datum rolvaliduntilDatum;
+	Datum rolpasswordDatum;
+	bool isNull = true;
+	Form_pg_authid role = ((Form_pg_authid) GETSTRUCT(tuple));
+	AlterRoleStmt *stmt = makeNode(AlterRoleStmt);
+
+	stmt->role = makeNode(RoleSpec);
+	stmt->role->roletype = ROLESPEC_CSTRING;
+	stmt->role->location = -1;
+	stmt->role->rolename = pstrdup(NameStr(role->rolname));
+	stmt->action = 1;
+	stmt->options = NIL;
+
+	stmt->options = lappend(stmt->options, makeDefElem("superuser", (Node *) makeInteger(
+														   role->rolsuper), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("createdb", (Node *) makeInteger(
+														   role->rolcreatedb), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("createrole", (Node *) makeInteger(
+														   role->rolcreaterole), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("inherit", (Node *) makeInteger(
+														   role->rolinherit), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("canlogin", (Node *) makeInteger(
+														   role->rolcanlogin), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("isreplication",
+													   (Node *) makeInteger(
+														   role->rolreplication), -1));
+	stmt->options = lappend(stmt->options, makeDefElem("bypassrls", (Node *) makeInteger(
+														   role->rolbypassrls), -1));
+
+	stmt->options = lappend(stmt->options, makeDefElem("connectionlimit",
+													   (Node *) makeInteger(
+														   role->rolconnlimit), -1));
+
+	rolpasswordDatum = heap_getattr(tuple, Anum_pg_authid_rolpassword,
+									pgAuthIdDescription, &isNull);
+	if (!isNull)
+	{
+		rolpassword = TextDatumGetCString(rolpasswordDatum);
+		stmt->options = lappend(stmt->options, makeDefElem("password",
+														   (Node *) makeString(
+															   rolpassword), -1));
+	}
+	else
+	{
+		stmt->options = lappend(stmt->options, makeDefElem("password", NULL, -1));
+	}
+
+	rolvaliduntilDatum = heap_getattr(tuple, Anum_pg_authid_rolvaliduntil,
+									  pgAuthIdDescription, &isNull);
+	if (!isNull)
+	{
+		rolvaliduntil = (char *) timestamptz_to_str(rolvaliduntilDatum);
+	}
+
+	stmt->options = lappend(stmt->options, makeDefElem("validUntil", (Node *) makeString(
+														   rolvaliduntil), -1));
+
+	return CreateAlterRoleIfExistsCommand(stmt);
+}
+
+
+/*
+ * GenerateAlterRoleIfExistsCommandAllRoles creates the set password queries
+ * for users whose password exists in the pg_authid table.
+ */
+List *
+GenerateAlterRoleIfExistsCommandAllRoles()
+{
+	Relation pgAuthId = heap_open(AuthIdRelationId, AccessShareLock);
+	TupleDesc pgAuthIdDescription = RelationGetDescr(pgAuthId);
+	HeapTuple tuple = NULL;
+	List *commands = NIL;
+	const char *alterRoleQuery = NULL;
+
+#if PG_VERSION_NUM >= 120000
+	TableScanDesc scan = table_beginscan_catalog(pgAuthId, 0, NULL);
+#else
+	HeapScanDesc scan = heap_beginscan_catalog(pgAuthId, 0, NULL);
+#endif
+
+	while ((tuple = heap_getnext(scan, ForwardScanDirection)) != NULL)
+	{
```

Comment:
minor: assingning `NameStr(((Form_pg_authid) GETSTRUCT(tuple))->rolname` to a  local  variable  might b e useful